locksmith/src/controllers/v2/metadataController.ts

```diff
@@ -416,30 +416,39 @@ export class MetadataController {
         return await metadataOperations.getKeyCentricData(lockAddress, keyId)
       })
 
+      // get owners address from `ownerOf`, because not every item can include the user address, and void undefined values
+      const ownersAddressPromise = keyIds.map(async (keyId) => {
```

Comment:
I think this is going to make the request timeout because looking up the owner for each of them will be a problem.

Reply:
refactored to avoid this issue